app/js/controllers/main.js

```diff
@@ -38,48 +32,23 @@ export default class MainController extends Controller {
       },
       maxTime: 1000,    // in less than 1 second.
     };
-
-    Gesture.detect(closeGesture).then(() => {
-      this.removeView();
-      this.waitToBeOpened();
-    });
   }
 
-  attachView() {
-    if (this.attached) {
-      return;
-    }
-
-    this.view.customizer.setRootNode(document.documentElement);
-    document.body.appendChild(this.view.el);
-
-    var padding = this.view.el.querySelector('fxos-customizer').offsetHeight;
-    this._styleEl = document.createElement('style');
-    this._styleEl.textContent = `
-    html, body {
-      overflow: initial !important;
-    }
-
-    body {
-      padding-bottom: ${padding}px !important;
-    }`;
-    document.head.appendChild(this._styleEl);
-
-    this.attached = true;
+  // These private _wait methods call each other. If you try to call them
+  // directly, you'll end up with more than one gesture detector listening
+  // at a time.
+  _waitToBeOpened() {
+    Gesture.detect(this.openGesture)
+      .then(() => this.view.open())
+      .then(() => {
+        this.view.customizer.setRootNode(document.documentElement);
+        this._waitToBeClosed();
```

Comment:
Nit-picking: it's weird how this is broken up into two ```then()``` blocks as the whole thing could really execute as one block, I think. Maybe I'm missing something, though.